### PR TITLE
search: expose query printers via GQL

### PR DESCRIPTION
**Pull request by rvantonder**
It's useful to expose our printer output via GQL for debugging. This is proposed schema changes. Thoughts? I'll implement the logic if we're happy with this.

## Test plan
Just schema change right now.

### CODE REVIEW

File: cmd/frontend/graphqlbackend/schema.graphql [hunk 1]
```diff
@@ -1932,6 +1944,54 @@ enum SearchPatternType {
...
+     """
+     JSON
+     """
+     S-expression format.
+     """
+     SEXP
+     """
+     Mermaid flowchart format.
+     """
+     MERMAID
```
Comment (rvantonder) [inline cmd/frontend/graphqlbackend/schema.graphql:1992]: We don't support these output formats for `PARSE_TREE` yet but that's an implementation case I'll handle.

**Review by camdencheek — APPROVED** [review]
LGTM. Would definitely be useful for debugging customer issues.

**Review by tbliu98 — APPROVED** [review]
I like it. I'm not familiar with consumers of the GQL API but I definitely see this being useful

**Comment by rvantonder** [comment]
only updated resolver to accept these values, so will merge as-is and implement logic in follow up

**Pull request merged**